src/SurfaceFluxes.jl

```diff
@@ -401,13 +446,13 @@ function obukhov_length(
         # Return ζ->0 in the neutral boundary layer case, where ζ = z / L_MO
         return L_MO = FT(Inf * sign(non_zero(ΔDSEᵥ)))
     else
-        function root_l_mo(x_lmo)
-            residual = x_lmo - local_lmo(param_set, x_lmo, sc, uft, scheme)
+        function root_ζ(ζ)
+            residual = compute_richardson_number(sc, DSEᵥ_in, DSEᵥ_sfc, grav) - compute_Ri_b(param_set,sc, uft, scheme, non_zero(ζ))
             return residual
         end
-        L_MO_init = FT(-1)
-        sol = RS.find_zero(root_l_mo, RS.NewtonsMethodAD(L_MO_init), soltype, tol, maxiter)
-        L_MO = sol.root
+        ζ₀ = FT(sign(ΔDSEᵥ))
+        sol = RS.find_zero(root_ζ, RS.NewtonsMethodAD(ζ₀), soltype, tol, maxiter)
+        L_MO = sol.root * Δz(sc)
```

Comment:
```suggestion
        L_MO = Δz(sc) / sol.root
```
It should be like this I think?

Reply:
yup zeta = z/Lmo. typo fixed.